pkgs/development/tools/godot/mono/make-deps.nix

```diff
@@ -0,0 +1,56 @@
+{ godot-mono, lib, nuget-to-nix }:
+
+# This can be used to recreate deps.nix. Because this entails doing
+# things that are not permitted within a sandboxed Nix build, this is
+# not done by building the derivation. Instead, a development shell
+# based on this derivation should be used to run the makeDepsPhase
+# script provided below.
+#
+# This can be done in the following ways:
+#
+# $ nix-shell <path-to-nixpkgs> -A godot-mono.make-deps --run 'eval "$makeDepsPhase"'
+#
+# or:
+#
+# $ nix develop <path-to-nixpkgs>#godot-mono.make-deps
+# $ eval "$makeDepsPhase"
+#
+# If successful, the final line of output will provide the path to the deps.nix in a temp folder.
+
+godot-mono.overrideAttrs (self: base: {
```

Comment:
Instead of exposing this derivation directly and asking the user to execute this manually; why not create a script (with writeScriptBin) that just has `nix-shell --pure ${pkgs.path} -A godot-mono.make-deps --run 'eval "$makeDepsPhase"'`? This way the user can just run a script instead of manually executing the step, which should be less error prone.

As a full example:

move the current make-deps.nix to make-deps-drv.nix

```nix
# make-deps.nix
{writeScriptBin, pkgs, stdenv}:
writeScriptBin "generate-nuget-deps" ''
  #!${stdenv.shell}
  nix-shell --pure ${pkgs.path} -A godot-mono.make-deps-drv --run 'eval "$makeDepsPhase"'
''
```

in `mono/default.nix`:

```nix
{
  # ...
  passthru = {
    # ...
    make-deps-drv = callPackage ./make-deps-drv.nix {};
    make-deps = callPackage ./make-deps.nix {};
    # ...
  };
  # ...
}
```

Reply:
How's this?

Reply:
I don't think it's really hacky. Looks fine IMO.